src/tests/Button.test.js

```diff
@@ -1,14 +1,41 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
-import Enzyme, { shallow } from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
+import Enzyme, { shallow, render, mount } from 'enzyme';
 import Button from '../components/Button/Button';
 
-Enzyme.configure({ adapter: new Adapter() });
 
 describe('<Button />', () => {
+  const button = shallow(<Button />);
+
+
   it('renders without crashing', () => {
-    const div = document.createElement('div');
-    ReactDOM.render(<Button />, div);
+    render(button);
+  });
+
+  it('have a button', () => {
+    button.find("component-button");
+    expect(button).toHaveLength(1);
+  });
+
+  it("have a name prop", () => {
+    const props = {name:'9'};
+    const bouton = shallow(<Button {...props}/>);
+    const button = bouton.find('h3');
+
+    expect(button.text()).toEqual('9');
+  })
+
+
+
+  it("have a button click", () => {
+    const HandleChange = jest.fn();
```

Comment:
I like that you took the time to pass the jest.fn spy function into the button to spy into its internal functionality, then simulated a click event to verify the spy function call times. Alternatively you can also check what arguments were passed to the function if any.